table/block_based_table_builder.cc

```diff
@@ -101,83 +101,123 @@ bool GoodCompressionRatio(size_t compressed_size, size_t raw_size) {
-Slice CompressBlock(const Slice& raw, const CompressionContext& compression_ctx,
-                    CompressionType* type, uint32_t format_version,
-                    std::string* compressed_output) {
-  *type = compression_ctx.type();
-  if (compression_ctx.type() == kNoCompression) {
-    return raw;
-  }
-
+bool CompressBlockInternal(
+    const Slice& raw, const CompressionContext& compression_ctx,
+    CompressionType type, uint32_t format_version,
+    std::string* compressed_output) {
   // Will return compressed block contents if (1) the compression method is
   // supported in this platform and (2) the compression rate is "good enough".
-  switch (compression_ctx.type()) {
+  switch (type) {
     case kSnappyCompression:
-      if (Snappy_Compress(compression_ctx, raw.data(), raw.size(),
-                          compressed_output) &&
-          GoodCompressionRatio(compressed_output->size(), raw.size())) {
-        return *compressed_output;
-      }
-      break;  // fall back to no compression.
+      return Snappy_Compress(
+          compression_ctx, raw.data(), raw.size(), compressed_output);
     case kZlibCompression:
-      if (Zlib_Compress(
-              compression_ctx,
-              GetCompressFormatForVersion(kZlibCompression, format_version),
-              raw.data(), raw.size(), compressed_output) &&
-          GoodCompressionRatio(compressed_output->size(), raw.size())) {
-        return *compressed_output;
-      }
-      break;  // fall back to no compression.
+      return Zlib_Compress(
+          compression_ctx,
+          GetCompressFormatForVersion(kZlibCompression, format_version),
+          raw.data(),
+          raw.size(),
+          compressed_output);
     case kBZip2Compression:
-      if (BZip2_Compress(
-              compression_ctx,
-              GetCompressFormatForVersion(kBZip2Compression, format_version),
-              raw.data(), raw.size(), compressed_output) &&
-          GoodCompressionRatio(compressed_output->size(), raw.size())) {
-        return *compressed_output;
-      }
-      break;  // fall back to no compression.
+      return BZip2_Compress(
+          compression_ctx,
+          GetCompressFormatForVersion(kBZip2Compression, format_version),
+          raw.data(),
+          raw.size(),
+          compressed_output);
     case kLZ4Compression:
-      if (LZ4_Compress(
-              compression_ctx,
-              GetCompressFormatForVersion(kLZ4Compression, format_version),
-              raw.data(), raw.size(), compressed_output) &&
-          GoodCompressionRatio(compressed_output->size(), raw.size())) {
-        return *compressed_output;
-      }
-      break;  // fall back to no compression.
+      return LZ4_Compress(
+          compression_ctx,
+          GetCompressFormatForVersion(kLZ4Compression, format_version),
+          raw.data(),
+          raw.size(),
+          compressed_output);
```

Comment:
Is it done by "make format"? If you haven't done it yet, run it and see.

Reply:
Looking at it.